mlir/unittests/Dialect/PDL/BuiltinTest.cpp

```diff
@@ -182,4 +398,101 @@ TEST_F(BuiltinTest, add) {
     EXPECT_TRUE(builtin::add(rewriter, results, {zerof32, zerof64}).failed());
   }
 }
+
+TEST_F(BuiltinTest, sub) {
```

Comment:
Will it make sense to properly test every data type (or at least the types we are using now) like [s/u]i8, [s/u]i32, [s/u]i64, f16, f32, f64 for corner cases (overflow, underflow, division by zero, ...) and correctness? or you believe because you are using [u/s]mul/rem/udiv/ from APINT/APfloat, it's not necessary.
In anycase, please add a test for division by zero.